### PR TITLE
refactor, chore(egen): Removes boilerplate, heading fixes, and other corrections from template

### PR DESCRIPTION
**Pull request by nileshspringml**
**REQUIRED:** 

- Removed boilar code
- Changed REGION variable name to LOCATION
- Added code for removing local generated files while executing code
- Changed default hardware from GPU to CPU
- Added extra code for using docker image and did major changes
- Added code for cleanup
- Colab enterpise logo added
- Written bullet points into different line. As currently they written on single line

<br>
Did major changes related to docker code and adapted notebook with new notebook template
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [ ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

2. If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/community/CODEOWNERS) file under the `Community Notebooks` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

<br>

3. If you are opening a PR for `Community Content` under the [community-content](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/community-content) folder:
- [ ] Make sure your main `Content Directory Name` is descriptive, informative, and includes some of the key products and attributes of your content, so that it is differentiable from other content
- [ ] The main content directory has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/community-content/CODEOWNERS) file under the `Community Content` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

### CODE REVIEW

File: notebooks/official/custom/custom_training_container_and_model_registry.ipynb [hunk 1]
```diff
@@ -401,18 +334,18 @@
         "\n",
         "Otherwise specify `(None, None)` to use a container image to run on a CPU.\n",
         "\n",
-        "Learn more [here](https://cloud.google.com/vertex-ai/docs/general/locations#accelerators) hardware accelerator support for your region"
+        "Learn more [here](https://cloud.google.com/vertex-ai/docs/general/locations#accelerators) hardware accelerator support for your location"
```
Comment (kittyabs) [inline notebooks/official/custom/custom_training_container_and_model_registry.ipynb:337]: Don't use "here" as link text. Recommend changing to:

- Learn more about [hardware accelerator support ](https://cloud.google.com/vertex-ai/docs/general/locations#accelerators) for your location"

File: notebooks/official/custom/custom_training_container_and_model_registry.ipynb [hunk 2]
```diff
@@ -749,107 +720,316 @@
...
+        "id": "a2ed53b6e259"
       },
       "source": [
-        "#### Build the container locally\n",
+        "### Enable Artifact Registry API\n",
         "\n",
-        "Next, you will provide a name for your customer container that you will use when you submit it to the Google Container Registry."
+        "First, you must enable the Artifact Registry API service for your project.\n",
+        "\n",
+        "Learn more about [Enabling service](https://cloud.google.com/artifact-registry/docs/enable-service)."
```
Comment (kittyabs) [inline notebooks/official/custom/custom_training_container_and_model_registry.ipynb:730]: To improve parsing, change to:

- Learn more about [enabling the Artifact Registry service](https://cloud.google.com/artifact-registry/docs/enable-service).

File: notebooks/official/custom/custom_training_container_and_model_registry.ipynb [hunk 3]
```diff
@@ -749,107 +720,316 @@
...
       "metadata": {
-        "id": "build_container:training"
+        "id": "94f77f953470"
       },
-      "outputs": [],
       "source": [
-        "! docker build custom -t $TRAIN_IMAGE"
+        "### Configure authentication to your private repo\n",
+        "\n",
+        "Before you push or pull container images, configure Docker to use the `gcloud` command-line tool to authenticate requests to `Artifact Registry` for your region."
```
Comment (kittyabs) [inline notebooks/official/custom/custom_training_container_and_model_registry.ipynb:794]: remove back ticks from Artifact Registry

File: notebooks/official/custom/custom_training_container_and_model_registry.ipynb [hunk 4]
```diff
@@ -749,107 +720,316 @@
...
+        "cd .."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ffff1d8b23fd"
+      },
+      "source": [
         "Next, you package the training folder into a compressed tar ball, and then store it in your Cloud Storage bucket."
```
Comment (kittyabs) [inline notebooks/official/custom/custom_training_container_and_model_registry.ipynb:1004]: remove "you" and change "it" to "the package:

- Next, package the training folder into a compressed tar ball, and then store the package in your Cloud Storage bucket.

**Review by kittyabs — COMMENTED** [review]
suggested edits

File: notebooks/official/custom/custom_training_container_and_model_registry.ipynb (hunk 1 above)
Comment (nileshspringml) [reply notebooks/official/custom/custom_training_container_and_model_registry.ipynb]: Did required changes

File: notebooks/official/custom/custom_training_container_and_model_registry.ipynb (hunk 2 above)
Comment (nileshspringml) [reply notebooks/official/custom/custom_training_container_and_model_registry.ipynb]: Did required changes

File: notebooks/official/custom/custom_training_container_and_model_registry.ipynb (hunk 3 above)
Comment (nileshspringml) [reply notebooks/official/custom/custom_training_container_and_model_registry.ipynb]: Did required changes

File: notebooks/official/custom/custom_training_container_and_model_registry.ipynb (hunk 4 above)
Comment (nileshspringml) [reply notebooks/official/custom/custom_training_container_and_model_registry.ipynb]: Did required changes

**Review by katiemn — APPROVED** [review]

**Pull request merged**